test/ha/activator_test.go

```diff
@@ -142,3 +154,66 @@ func assertSLO(t *testing.T, p test.Prober, slo float64) {
 		t.Error("CheckSLO failed:", err)
 	}
 }
+
+type activatorPod struct {
+	name string
+	ip   string
+}
+
+func gatherBackingActivators(ctx context.Context, client kubernetes.Interface, namespace string, revs ...string) ([]activatorPod, error) {
+	podMap := make(map[string]activatorPod)
+
+	for _, rev := range revs {
+		endpoints := client.CoreV1().Endpoints(namespace)
+		e, err := endpoints.Get(ctx, rev, metav1.GetOptions{})
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to gather %s endpoints: %w", rev, err)
+		}
+
+		for _, subset := range e.Subsets {
+			for _, address := range subset.Addresses {
+				if address.TargetRef == nil {
+					return nil, fmt.Errorf("%s service is not pointing to a pod", rev)
+				}
+
+				name := address.TargetRef.Name
+				if !strings.HasPrefix(name, activatorDeploymentName) {
+					return nil, fmt.Errorf("%s service is not pointing to an activator pod but: %s", rev, address.TargetRef.Name)
+				}
+
+				podMap[name] = activatorPod{name: name, ip: address.IP}
+			}
+		}
+	}
+
+	pods := make([]activatorPod, 0, len(podMap))
+	for _, pod := range podMap {
+		pods = append(pods, pod)
+	}
+
+	return pods, nil
+}
+
+func waitForActivatorScale(ctx context.Context, client kubernetes.Interface) (int, error) {
+	desiredScale := 0
+	check := func(d *appsv1.Deployment) (bool, error) {
+		if *d.Spec.Replicas < 2 {
+			return false, errors.New("spec.replicaCount should be > 1")
+		}
+		desiredScale = int(*d.Spec.Replicas)
+		return d.Status.ReadyReplicas > 1, nil
```

Comment:
If we need this function to wait for a consistent scale, should this wait for ready replicas == spec.replicas?

Reply:
I'll just look at the deployment condition Available == True